python/ql/src/experimental/semmle/python/security/injection/RegexInjection.qll

```diff
@@ -0,0 +1,25 @@
+/**
+ * Provides a taint-tracking configuration for detecting regular expression injection
+ * vulnerabilities.
+ */
+
+import python
+import experimental.semmle.python.Concepts
+import experimental.semmle.python.frameworks.Stdlib
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.TaintTracking
+import semmle.python.dataflow.new.RemoteFlowSources
+import semmle.python.ApiGraphs
+
+/**
+ * A taint-tracking configuration for detecting regular expression injections.
+ */
+class RegexInjectionFlowConfig extends TaintTracking::Configuration {
+  RegexInjectionFlowConfig() { this = "RegexInjectionFlowConfig" }
+
+  override predicate isSource(DataFlow::Node source) { source instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) { sink = any(RegexExecution re).getRegexNode() }
+
+  override predicate isSanitizer(DataFlow::Node sanitizer) { sanitizer instanceof ReEscapeCall }
```

Comment:
How could we access `ReEscapeCall`'s `getRegexNode()`? Something like `sanitizer = any(ReEscapeCall foo).getRegexNode()`?

Reply:
You would do `sanitizer.(ReEscapeCall).getRegexNode()`. The parentheses perform a cast.

Reply:
Tried that but the class can't be found even though Stdlib is imported :o